tests/Microsoft.Identity.Test.Unit.net45/CacheTests/UnifiedSchemaValidationTests.cs

```diff
@@ -0,0 +1,842 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Identity.Client.Cache.Items;
+using Microsoft.Identity.Client.Cache.Keys;
+using Microsoft.Identity.Client.OAuth2;
+using Microsoft.Identity.Client.Utils;
+using Microsoft.Identity.Json.Linq;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Identity.Test.Unit.CacheTests
+{
+    // This is meant to stay in sync with iOS code in https://github.com/AzureAD/microsoft-authentication-library-common-for-objc/blob/master/IdentityCore/tests/integration/MSIDCacheSchemaValidationTests.m
+    // This is for tests laid out here:  https://identitydivision.visualstudio.com/DevEx/_git/AuthLibrariesApiReview?path=%2FUnifiedSchema%2Ftestcases&version=GBdev
+
+    [TestClass]
+    [TestCategory("UnifiedSchema_Validation")]
+    public class UnifiedSchemaValidationTests : TestBase
+    {
+        private const string ClientId = "b6c69a37-df96-4db0-9088-2ab96e1d8215";
+        private const string B2CClientId = "0a7f52dd-260e-432f-94de-b47828c3f372";
+        private const string B2CScopes = "https://iosmsalb2c.onmicrosoft.com/webapitest/user.read";
+        private const string AuthorityUri = "https://login.microsoftonline.com/common";
+        private const string MsalEnvironment = "login.microsoftonline.com";
```

Comment:
what about b2c and custom domains?

Reply:
this is for now just a replica of what the iOS team has.  i can talk to olga about adding tests here for custom domains and expanding b2c tests

Reply:
hmmm...interesting. just doesn't reflect what customers are using.

---
In reply to: [299257205](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1243#discussion_r299257205) [](ancestors = 299257205)